linode/manager
# feat: [M3-6755] - Unassign Linodes from Subnet

## Description 📝
Allow for un-assigning Linodes from subnets.

## Major Changes 🔄
- Allow for Linodes can be batch unassigned
- Add unassigned Linodes to CSV in order to provide list for instances that need to be rebooted

## Preview 
| Screenshots  | Video  |
| ------- | ------- |
| ![Screenshot 2023-09-26 at 1 42 39 PM](https://github.com/linode/manager/assets/125309814/0fcceea4-81ee-4b4a-a511-77abe0278492) | <video src="https://github.com/linode/manager/assets/125309814/5dea40cc-0919-4bf7-838d-5be8423b7d25" /> |

## How to test 🧪
> **Note**
Unit / Integration Tests will be handled in a separate PR [M3-6771]

1. Ensure you have both customer tags applied (see me for details)
2. Ensure feature flag is enabled in environment
3. Point `env` variables to DEV and use incognito to hit local host
4. Go to VPC > Subnets and ensure you have some Linode configs in your subnet (if not, add them)
5. Otherwise, click actions menu "..." to `Unassign Linodes`
6. Selecting a linode config should show "Linodes to be Unassigned from Subnet (x)" with a list of Linodes selecting
    1. You should be able to undo selections in multiple ways via the UI
8. Downloading the CSV (no linodes assigned, 1+ linodes assigned, determine that CSV accurately reflects the linodes in the 'Linodes assigned to Subnet (x) list) (should have linode's ID, label, public ipv4 at minimum, may include more fields)
9. Double check that you can't unassign Linodes to a subnet if you don't have the permissions for that specific vpc
